src/order/ideal.lean

```diff
@@ -122,6 +122,74 @@ I.mem_of_le (sup_le hx hy) h_mem
 
 end semilattice_sup
 
+section semilattice_sup_bot
+variables [semilattice_sup_bot P] (I J K : ideal P)
+
+/-- The intersection of two ideals is an ideal, when `P` has joins and a bottom. -/
+def inf (I J : ideal P) : ideal P :=
+{ carrier   := I ∩ J,
```

Comment:
There should also be a simp lemma `mem_inf : x ∈ I ⊓ J ↔ x ∈ I ∧ x ∈ J` as well as a lemma `mem_sup : x ∈ I ⊔ J ↔ ∃ (i ∈ I) (j ∈ J), x ≤ i ⊔ j`.